.github/workflows/rerun-flaky-workflows.yml

```diff
@@ -15,22 +15,34 @@ jobs:
       actions: write
     steps:
       - name: Display Workflow Run Information
+        env:
+          WORKFLOW_RUN_ID: ${{ github.event.workflow_run.id }}
+          WORKFLOW_RUN_NAME: ${{ github.event.workflow_run.name }}
+          WORKFLOW_RUN_ATTEMPT: ${{ github.event.workflow_run.run_attempt }}
+          WORKFLOW_RUN_HTML_URL: ${{ github.event.workflow_run.html_url }}
+          GITHUB_JOB: ${{ github.job }}
+          GITHUB_RUN_ATTEMPT: ${{ github.run_attempt }}
+          AUTHOR_NAME: ${{ github.event.workflow_run.head_commit.author.name }}
```

Comment:
This is a list of GitHub context inputs that should not be used directly when creating inline scripts:

- `github.event.comment.body`
- `github.event.issue.body`
- `github.event.issue.title`
- `github.head_ref`
- `github.pull_request.*`
- `github.*.*.author.name`
- `github.*.*.author.email`

According to the security hardening guidelines mentioned [here](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections), there is a risk associated with treating certain inputs as potential untrusted input, such as `author.name`. Based on best practices outlined [here](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable), I refactored this workflow.

It is possible to revert and only keep the required ones.